### PR TITLE
Test rendering

### PR DESCRIPTION
**Pull request by esc**
This re-enabled testing fig. 3 and fig. 4 from the Bahmann paper. The output must still be inspected by a human, but any program crashes due to exceptions would be noticed.

### CODE REVIEW

**Comment by esc** [comment]
Unfortunately, making the tests pass, resulted in follow-up failures 😭

**Comment by esc** [comment]
> Unfortunately, making the tests pass, resulted in follow-up failures 😭

The simulator tests could be fixed with a one-line change in [e852fb1](https://github.com/numba/numba-rvsdg/pull/39/commits/e852fb17e179f7483393cf62562d4c4676a6f88c)

**Comment by esc** [comment]
Coverage for the `rendering` module has increased to 92%

File: numba_rvsdg/tests/test_fig4.py [hunk 1]
```diff
@@ -30,3 +30,10 @@ def make_flow():
     flow = FlowInfo.from_bytecode(bc)
     bbmap = flow.build_basicblocks()
     return ByteFlow(bc=bc, bbmap=bbmap)
+
+def test_fig4():
+    render_flow(make_flow())
```
Comment (kc611) [inline numba_rvsdg/tests/test_fig4.py:35]: Question: Using `render_flow` here will probably lead to the graph every time `pytest` suite is run locally. Can we  have a functional version of this which keeps rendering optional ?

Something like:
```python
if render_graph:
   render_flow(make_flow())
else:
    flow = make_flow()
    cflow = flow._join_returns()
    lflow = cflow._restructure_loop()
    bflow = lflow._restructure_branch()
```

Where `render_graph` is an environment boolean.

**Review by kc611 — COMMENTED** [review]
Functional changes LGTM, one question below.

File: numba_rvsdg/tests/test_fig4.py (hunk 1 above)
Comment (esc) [reply numba_rvsdg/tests/test_fig4.py]: I added a07a567 with the following comment:

```
When running the test via `pytest` nothing is displayed. When running
the file with `python` on the command line, the output is rendered.
Thus, the files can be used during testing but also to visualize the
results.
```

**Comment by esc** [comment]
CI hit the Shroedinger-Bug which is  maybe fixed by #40

**Comment by esc** [comment]
> CI hit the Shroedinger-Bug which is maybe fixed by #40

After a re-run, CI passed.

**Pull request merged**